package.json

```diff
@@ -25,26 +25,25 @@
     "node": ">= 5.4.1"
   },
   "dependencies": {
-    "@angular/animations": "~4.4.4",
-    "@angular/common": "~4.4.4",
-    "@angular/compiler": "~4.4.4",
-    "@angular/core": "~4.4.4",
-    "@angular/forms": "~4.4.4",
-    "@angular/http": "~4.4.4",
-    "@angular/platform-browser": "~4.4.4",
+    "@angular/animations": "~5.0.0-rc.4",
+    "@angular/common": "~5.0.0-rc.4",
+    "@angular/compiler": "~5.0.0-rc.4",
+    "@angular/core": "~5.0.0-rc.4",
+    "@angular/forms": "~5.0.0-rc.4",
+    "@angular/http": "~5.0.0-rc.4",
+    "@angular/platform-browser": "~5.0.0-rc.4",
     "core-js": "^2.4.1",
-    "rxjs": "^5.0.1",
+    "rxjs": "5.x",
     "systemjs": "0.19.43",
-    "tsickle": "^0.23.5",
+    "tsickle": "^0.24.x",
     "tslib": "^1.7.1",
     "zone.js": "^0.8.12"
   },
   "devDependencies": {
-    "@angular/compiler-cli": "~4.4.4",
-    "@angular/platform-browser-dynamic": "~4.4.4",
-    "@angular/platform-server": "~4.4.4",
-    "@angular/router": "~4.4.4",
-    "@angular/tsc-wrapped": "~4.4.4",
```

Comment:
As material used to use `tsc-wrapped` directly, but is now using `ngc`, you need to make sure to add `skipTemplateCodegen` to all of your `tsconfig.json` files, including the ones that inherit from a base `tsconfig.json` as ngc does not yet support inheritance. 

E.g. `src/material-moment-adapter/tsconfig-tests.json` is missing this setting.

Reply:
Thanks, that sorted out the build error that I was running into. There's one more issue though: It seems like the compiler is dropping some of the imports (the ones that were purely used for type info). E.g. [this one](https://github.com/angular/material2/blob/master/src/cdk/overlay/overlay.ts#L21) is removed at compile time while the class' `ctorParameters` still reference it:

```js
Overlay.ctorParameters = function () { return [
    { type: index_1.ScrollStrategyOptions, },
    { type: overlay_container_1.OverlayContainer, },
    { type: core_1.ComponentFactoryResolver, },
    { type: overlay_position_builder_1.OverlayPositionBuilder, },
    { type: core_1.ApplicationRef, },
    { type: core_1.Injector, },
    { type: core_1.NgZone, },
]; };
```